src/node.cc

```diff
@@ -832,20 +832,21 @@ static ExitCode InitializeNodeWithArgsInternal(
   HandleEnvOptions(per_process::cli_options->per_isolate->per_env);
 
   std::string node_options;
-  auto file_paths = node::Dotenv::GetPathFromArgs(*argv);
+  auto env_files = node::Dotenv::GetEnvFileDataFromArgs(*argv);
 
-  if (!file_paths.empty()) {
+  if (!env_files.empty()) {
     CHECK(!per_process::v8_initialized);
 
-    for (const auto& file_path : file_paths) {
-      switch (per_process::dotenv_file.ParsePath(file_path)) {
+    for (const auto& file_data : env_files) {
+      switch (per_process::dotenv_file.ParsePath(file_data.path)) {
         case Dotenv::ParseResult::Valid:
           break;
         case Dotenv::ParseResult::InvalidContent:
-          errors->push_back(file_path + ": invalid format");
+          errors->push_back(file_data.path + ": invalid format");
           break;
         case Dotenv::ParseResult::FileError:
-          errors->push_back(file_path + ": not found");
+          if (!file_data.is_required) continue;
+          errors->push_back(file_data.path + ": not found");
```

Comment:
While this is a pre-existing bug, this really should not print `not found` if the file could not be opened due to an error other than `ENOENT`. (Unfortunately, `ParsePath()` just returns a generic `FileError` value regardless of the error code.)

Reply:
Good catch. How'd you suggest we handle this, simply not print anything? Improving the check as to why it errored in the first place? I do believe it should still error if the flag used was `--env-file`

Reply:
I think it's fine to leave it as it is for this PR since it's a pre-existing bug. Eventually, the code should check if the error code was `ENOENT`, and if it's not that, then print the appropriate error message for the error code.